constraints.go

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+package magistrala
+
+type Operation int
+
+const (
+	Create Operation = iota
+)
+
+// Constraints specifies an API for obtaining entity constraints.
+type Constraints interface {
+	// Constraints returns constraints for the entities.
+	CheckLimits(operation Operation, currentValue uint64) error
+}
```

Comment:
We should think about domains in constraint provider instead of passing current value. 
Current value should be retrieved by Check function itself
```suggestion
	Check(operation Operation, domainID string) error

```